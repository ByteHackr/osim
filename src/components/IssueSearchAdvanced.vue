<script setup lang="ts">
import { computed, onMounted, onUnmounted, ref, watch } from 'vue';

import { sort } from 'ramda';
// @ts-expect-error missing types
import DjangoQL from 'djangoql-completion';

import Modal from '@/components/widgets/Modal.vue';
import QueryFilterGuide from '@/components/QueryFilterGuide.vue';

import { useModal } from '@/composables/useModal';
import { useSearchParams } from '@/composables/useSearchParams';

import { flawImpacts, flawSources, flawIncidentStates, descriptionRequiredStates } from '@/types/zodFlaw';
import { flawFields } from '@/constants/flawFields';
import { affectAffectedness } from '@/types/zodAffect';
import { osimRuntime } from '@/stores/osimRuntime';

const props = defineProps<{
  isLoading: boolean;
}>();
const emit = defineEmits(['filter:save']);
const { facets, query, removeFacet, submitAdvancedSearch } = useSearchParams();
const { closeModal, isModalOpen, openModal } = useModal();

const queryFilterVisible = ref(true);

const emptinessSupportedFields = ['cve_id', 'cve_description', 'statement', 'mitigation', 'owner', 'cwe_id'];

const djangoCompletion = ref();

const nameForOption = (fieldName: string) => {
  const mappings: Record<string, string> = {
    uuid: 'UUID',
    cvss_scores__score: 'CVSS Score',
    cvss_scores__vector: 'CVSS Vector',
    affects__ps_module: 'Affected Module',
    affects__ps_component: 'Affected Component',
    affects__affectedness: 'Affected Affectedness',
    affects__trackers__ps_update_stream: 'Affect Update Stream',
    acknowledgments__name: 'Acknowledgment Author',
    affects__trackers__errata__advisory_name: 'Errata Advisory Name',
    affects__trackers__external_system_id: 'Tracker External System ID',
    workflow_state: 'Flaw State',
    cwe_id: 'CWE ID',
    cve_id: 'CVE ID',
    source: 'CVE Source',
    cve_description: 'CVE Description',
    requires_cve_description: 'CVE Description Review',
    major_incident_state: 'Incident State',
  };
  let name =
    mappings[fieldName]
    || fieldName.replace(/__[a-z]/g, label => `: ${label.charAt(2).toUpperCase()}`);
  name = name.replace(/_/g, ' ');
  return name.charAt(0).toUpperCase() + name.slice(1);
};

const sortFieldNames = sort((fieldA: string, fieldB: string) =>
  nameForOption(fieldA).localeCompare(nameForOption(fieldB)),
);

const chosenFields = computed(() => facets.value.map(({ field }) => field));

const unchosenFields = (chosenField: string) =>
  sortFieldNames(flawFields.filter(field => !chosenFields.value.includes(field) || field === chosenField));

const optionsFor = (field: string) =>
  ({
    source: flawSources,
    impact: flawImpacts,
    requires_cve_description: descriptionRequiredStates,
    embargoed: ['true', 'false'],
    workflow_state: [
      'DONE',
      'NEW',
      'PRE_SECONDARY_ASSESSMENT',
      'REJECTED',
      'SECONDARY_ASSESSMENT',
      'TRIAGE',
    ],
    major_incident_state: flawIncidentStates,
    affects__affectedness: affectAffectedness,
  })[field] || null;

const shouldShowAdvanced = ref(true);

watch(queryFilterVisible, () => {
  if (!queryFilterVisible.value) {
    query.value = '';
  }
});

onMounted(() => {
  const djangoQL = new DjangoQL({
    completionEnabled: true,
    introspections: `${osimRuntime.value.backends.osidb}/osidb/api/v1/introspection`,
    selector: 'textarea#query',
    autoResize: true,
    syntaxHelp: null,
    onSubmit: function (value: string) {
      query.value = value;
      submitAdvancedSearch();
    },
  });

  djangoQL.popupCompletion();

  djangoCompletion.value = djangoQL;
});

onUnmounted(() => {
  if (djangoCompletion.value) {
    djangoCompletion.value.destroyCompletionElement();
  }
});
</script>

<template>
  <details :open="shouldShowAdvanced" class="osim-advanced-search-container container-fluid">
    <summary class="mb-1" @click="shouldShowAdvanced = true">Advanced Search</summary>
    <form class="mb-2" @submit.prevent="submitAdvancedSearch">
      <div v-if="queryFilterVisible" class="input-group my-1">
        <div type="button" class="form-control bg-secondary text-white pe-none py-0 d-flex" style="max-width: 13.125%;">
          <span class="my-auto">Query Filter</span>
          <button class="btn btn-sm p-0 ms-auto my-auto text-white pe-auto border-0" type="button" @click="openModal()">
            <i class="bi bi-question-circle-fill fs-5" aria-label="hide query filter" />
          </button>
        </div>
        <textarea
          id="query"
          v-model="query"
          type="text"
          class="form-control"
        />
        <button
          class="btn btn-sm btn-secondary rounded-0 py-0"
          title="Clear query"
          type="button"
          @click="query = ''"
        >
          <i class="bi bi-eraser fs-5"></i>
        </button>
        <button class="btn btn-secondary py-0" type="button" @click="() => queryFilterVisible = false">
          <i class="bi bi-x fs-5" aria-label="hide query filter" />
        </button>
      </div>
      <div v-for="(facet, index) in facets" :key="facet.field" class="input-group my-1">
        <select v-model="facet.field" class="form-select search-facet-field" @submit.prevent>
          <option
            v-if="!facet.field"
            selected
            value=""
            disabled
          >Select field...</option>
          <option v-for="field in unchosenFields(facet.field)" :key="field" :value="field">
            {{ nameForOption(field) }}
          </option>
        </select>
        <select
          v-if="optionsFor(facet.field)"
          v-model="facet.value"
          class="form-select"
          @submit.prevent
        >
          <option v-for="option in optionsFor(facet.field)" :key="option" :value="option">
            {{ option }}
          </option>
        </select>
        <input
          v-else
          v-model="facet.value"
          type="text"
          class="form-control"
          :disabled="!facet.field"
        />
        <div v-if="emptinessSupportedFields.includes(facet.field)" class="btn-group">
          <button
            class="btn btn-sm btn-primary rounded-0 py-0"
            title="Non empty field search"
            type="button"
            :disabled="facet.value === 'isempty'"
            @click="facet.value = 'isempty'"
          >
            <i class="bi bi-circle fs-5" />
          </button>
          <button
            class="btn btn-sm btn-primary rounded-0 py-0"
            title="Empty field search"
            type="button"
            :disabled="facet.value === 'nonempty'"
            @click="facet.value = 'nonempty'"
          >
            <i class="bi bi-slash-circle fs-5" />
          </button>
        </div>
        <button
          class="btn btn-sm btn-primary rounded-0 py-0"
          title="Clear field"
          type="button"
          @click="facet.value = ''"
        >
          <i class="bi bi-eraser fs-5"></i>
        </button>
        <button class="btn btn-primary py-0" type="button" @click="removeFacet(index)">
          <i class="bi-x fs-5" aria-label="remove field"></i>
        </button>
      </div>
      <div class="mt-2">
        <button class="btn btn-primary me-2" type="submit" :disabled="props.isLoading">
          <div v-if="props.isLoading" class="spinner-border spinner-border-sm"></div>
          Search
        </button>
        <button
          class="btn btn-primary me-2"
          type="button"
          :disabled="isLoading"
          @click="emit('filter:save')"
        >
          Save as Default
        </button>
        <button
          v-if="!queryFilterVisible"
          class="btn btn-secondary"
          type="button"
          @click="() => queryFilterVisible = true"
        >
          Show Query Filter
        </button>
      </div>
    </form>
  </details>
  <Modal :show="isModalOpen" style="max-width: 75%;" @close="closeModal()">
    <template #header>
      <h1>Query Filter Guide</h1>
      <button
        type="button"
        class="btn-close ms-auto"
        aria-label="Close"
        @click="closeModal()"
      />
    </template>
    <template #body>
      <QueryFilterGuide />
    </template>
    <template #footer>
      <button
        type="button"
        class="btn btn-secondary m-0"
        @click="closeModal()"
      >
        Close
      </button>
    </template>
  </Modal>
</template>

<style lang="scss" scoped>
.osim-advanced-search-container {
  width: 97.5%;

  i {
    cursor: pointer;
  }

  select.search-facet-field {
    display: flex;
    width: auto;
    flex-grow: 0;
  }
}
</style>

<style lang="scss">
@import '@/scss/djangoql';
@import 'djangoql-completion/dist/completion.css';
</style>
