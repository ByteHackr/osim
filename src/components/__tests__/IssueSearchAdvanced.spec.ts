import { mount, VueWrapper, flushPromises } from '@vue/test-utils';
import { describe, it, expect, vi } from 'vitest';
import { useRoute, useRouter } from 'vue-router';
import IssueSearchAdvanced from '@/components/IssueSearchAdvanced.vue';

vi.mock('vue-router', async () => {
  const actual = await vi.importActual('vue-router');
  const replaceMock = vi.fn();
  
  return {
    ...actual,
    useRoute: vi.fn(() => ({ query: { mode: 'advanced', query: 'search' } })),
    useRouter: vi.fn(() => ({
      replace: replaceMock
    }))
  };
});


describe('IssueSearchAdvanced', () => {
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    const props: typeof IssueSearchAdvanced.props = { isLoading: false };
    wrapper = mount(IssueSearchAdvanced, {
      props,
      global: {
        mocks: { useRoute, useRouter },
      }
    });
  });

  it('should render', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should show advanced search', async () => {
    const content = wrapper.find('details');
    expect(content.exists()).toBe(true);
  });

  it('should update router with filters on search button click', async () => {
    const selectDropdown = wrapper.find('select.form-select.search-facet-field');
    await selectDropdown.setValue(selectDropdown.findAll('option')[1].element.value);
    await selectDropdown.trigger('change');
    const inputField = wrapper.find('input.form-control');
    await inputField.setValue('test'); 
    const searchButton = wrapper.find('button[type="submit"]');
    expect(searchButton.exists()).toBeTruthy();
    await searchButton.trigger('submit');
    await flushPromises();
    expect(useRouter().replace).toHaveBeenCalled();
    expect(useRouter().replace.mock.calls[0][0])
      .toStrictEqual({ query: { query: 'search', acknowledgments__name: 'test' } });
  });

  it('shouldn\'t render duplicate options on dropdown', () => {
    const selectDropdown = wrapper.find('select.form-select.search-facet-field');
    const allOptionsEL = selectDropdown.findAll('option');
    expect(allOptionsEL.length).toBe(21);
    const allValues = allOptionsEL.map(item => item.element.value);
    const uniqueValues = [...new Set(allValues)];
    expect(allValues).toStrictEqual(uniqueValues);
  });
});
