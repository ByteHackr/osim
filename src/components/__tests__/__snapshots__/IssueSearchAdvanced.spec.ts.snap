// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`issueSearchAdvanced > should add new facet when last facet is populated 1`] = `
"<details data-v-21c9158a="" open="" class="osim-advanced-search-container container-fluid">
  <summary data-v-21c9158a="" class="mb-1">Advanced Search</summary>
  <form data-v-21c9158a="" class="mb-2">
    <div data-v-21c9158a="" class="input-group my-1">
      <div data-v-21c9158a="" type="button" class="query-input form-control bg-secondary text-white"><span data-v-21c9158a="" class="my-auto">Query Filter</span><button data-v-21c9158a="" class="btn btn-sm text-white" type="button"><i data-v-21c9158a="" class="bi bi-question-circle-fill fs-5" aria-label="hide query filter"></i></button></div><textarea data-v-21c9158a="" id="query" type="text" class="form-control"></textarea><button data-v-21c9158a="" disabled="" class="btn btn-sm btn-secondary rounded-0 py-0" title="Clear query" type="button"><i data-v-21c9158a="" class="bi bi-eraser fs-5"></i></button><button data-v-21c9158a="" class="btn btn-secondary py-0" type="button"><i data-v-21c9158a="" class="bi bi-x fs-5" aria-label="hide query filter"></i></button>
    </div>
    <div data-v-21c9158a="" class="input-group my-1"><select data-v-21c9158a="" class="form-select search-facet-field">
        <!--v-if-->
        <option data-v-21c9158a="" value="acknowledgments__name">Acknowledgment Author</option>
        <option data-v-21c9158a="" value="affects__trackers__ps_update_stream">Affect Update Stream</option>
        <option data-v-21c9158a="" value="affects__affectedness">Affected Affectedness</option>
        <option data-v-21c9158a="" value="affects__ps_component">Affected Component</option>
        <option data-v-21c9158a="" value="affects__ps_module">Affected Module</option>
        <option data-v-21c9158a="" value="cve_description">CVE Description</option>
        <option data-v-21c9158a="" value="requires_cve_description">CVE Description Review</option>
        <option data-v-21c9158a="" value="cve_id">CVE ID</option>
        <option data-v-21c9158a="" value="source">CVE Source</option>
        <option data-v-21c9158a="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-21c9158a="" value="cvss_scores__vector">CVSS Vector</option>
        <option data-v-21c9158a="" value="cwe_id">CWE ID</option>
        <option data-v-21c9158a="" value="embargoed">Embargoed</option>
        <option data-v-21c9158a="" value="affects__trackers__errata__advisory_name">Errata Advisory Name</option>
        <option data-v-21c9158a="" value="workflow_state">Flaw State</option>
        <option data-v-21c9158a="" value="impact">Impact</option>
        <option data-v-21c9158a="" value="major_incident_state">Incident State</option>
        <option data-v-21c9158a="" value="mitigation">Mitigation</option>
        <option data-v-21c9158a="" value="owner">Owner</option>
        <option data-v-21c9158a="" value="statement">Statement</option>
        <option data-v-21c9158a="" value="title">Title</option>
        <option data-v-21c9158a="" value="affects__trackers__external_system_id">Tracker External System ID</option>
        <option data-v-21c9158a="" value="uuid">UUID</option>
      </select><input data-v-21c9158a="" type="text" class="form-control">
      <div data-v-21c9158a="" class="btn-group"><button data-v-21c9158a="" class="btn btn-sm btn-primary rounded-0 py-0" title="Non empty field search" type="button"><i data-v-21c9158a="" class="bi bi-circle fs-5"></i></button><button data-v-21c9158a="" class="btn btn-sm btn-primary rounded-0 py-0" title="Empty field search" type="button"><i data-v-21c9158a="" class="bi bi-slash-circle fs-5"></i></button></div><button data-v-21c9158a="" class="btn btn-sm btn-primary rounded-0 py-0" title="Clear field" type="button"><i data-v-21c9158a="" class="bi bi-eraser fs-5"></i></button><button data-v-21c9158a="" class="btn btn-primary py-0" type="button"><i data-v-21c9158a="" class="bi-x fs-5" aria-label="remove field"></i></button>
    </div>
    <div data-v-21c9158a="" class="input-group my-1"><select data-v-21c9158a="" class="form-select search-facet-field">
        <option data-v-21c9158a="" selected="" disabled="" value="">Select field...</option>
        <option data-v-21c9158a="" value="acknowledgments__name">Acknowledgment Author</option>
        <option data-v-21c9158a="" value="affects__trackers__ps_update_stream">Affect Update Stream</option>
        <option data-v-21c9158a="" value="affects__affectedness">Affected Affectedness</option>
        <option data-v-21c9158a="" value="affects__ps_component">Affected Component</option>
        <option data-v-21c9158a="" value="affects__ps_module">Affected Module</option>
        <option data-v-21c9158a="" value="requires_cve_description">CVE Description Review</option>
        <option data-v-21c9158a="" value="cve_id">CVE ID</option>
        <option data-v-21c9158a="" value="source">CVE Source</option>
        <option data-v-21c9158a="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-21c9158a="" value="cvss_scores__vector">CVSS Vector</option>
        <option data-v-21c9158a="" value="cwe_id">CWE ID</option>
        <option data-v-21c9158a="" value="embargoed">Embargoed</option>
        <option data-v-21c9158a="" value="affects__trackers__errata__advisory_name">Errata Advisory Name</option>
        <option data-v-21c9158a="" value="workflow_state">Flaw State</option>
        <option data-v-21c9158a="" value="impact">Impact</option>
        <option data-v-21c9158a="" value="major_incident_state">Incident State</option>
        <option data-v-21c9158a="" value="mitigation">Mitigation</option>
        <option data-v-21c9158a="" value="owner">Owner</option>
        <option data-v-21c9158a="" value="statement">Statement</option>
        <option data-v-21c9158a="" value="title">Title</option>
        <option data-v-21c9158a="" value="affects__trackers__external_system_id">Tracker External System ID</option>
        <option data-v-21c9158a="" value="uuid">UUID</option>
      </select><input data-v-21c9158a="" type="text" class="form-control" disabled="">
      <!--v-if-->
      <!--v-if--><button data-v-21c9158a="" class="btn btn-primary py-0" type="button"><i data-v-21c9158a="" class="bi-x fs-5" aria-label="remove field"></i></button>
    </div>
    <div data-v-21c9158a="" class="d-flex mt-2"><button data-v-21c9158a="" class="btn btn-primary me-2" type="submit" aria-label="Advance search button">
        <!--v-if--> Search
      </button><button data-v-21c9158a="" class="btn btn-primary me-2" aria-label="Save filters as default" type="button"> Save as Default </button>
      <!--v-if--><select data-v-21c9158a="" class="sort-by-select form-select search-facet-field" title="When using this sorting with active table column sort, the table column field will be used as secondary sorting key">
        <option data-v-21c9158a="" selected="" disabled="" label="Sort By..."></option>
        <option data-v-21c9158a="" hidden="" label="" value=""></option>
        <option data-v-21c9158a="" value="cve_description">CVE Description</option>
        <option data-v-21c9158a="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-21c9158a="" value="cwe_id">CWE ID</option>
        <option data-v-21c9158a="" value="major_incident_state">Incident State</option>
        <option data-v-21c9158a="" value="mitigation">Mitigation</option>
        <option data-v-21c9158a="" value="source">CVE Source</option>
        <option data-v-21c9158a="" value="statement">Statement</option>
      </select>
      <!--v-if-->
    </div>
  </form>
</details>
<transition-stub data-v-a347fd2c="" name="modal" appear="false" persisted="false" css="true">
  <!--v-if-->
</transition-stub>
<transition-stub data-v-a347fd2c="" name="modal-bg" appear="false" persisted="false" css="true">
  <!--v-if-->
</transition-stub>"
`;

exports[`issueSearchAdvanced > should render when \`isLoading\` is false 1`] = `
"<details data-v-21c9158a="" open="" class="osim-advanced-search-container container-fluid">
  <summary data-v-21c9158a="" class="mb-1">Advanced Search</summary>
  <form data-v-21c9158a="" class="mb-2">
    <div data-v-21c9158a="" class="input-group my-1">
      <div data-v-21c9158a="" type="button" class="query-input form-control bg-secondary text-white"><span data-v-21c9158a="" class="my-auto">Query Filter</span><button data-v-21c9158a="" class="btn btn-sm text-white" type="button"><i data-v-21c9158a="" class="bi bi-question-circle-fill fs-5" aria-label="hide query filter"></i></button></div><textarea data-v-21c9158a="" id="query" type="text" class="form-control"></textarea><button data-v-21c9158a="" disabled="" class="btn btn-sm btn-secondary rounded-0 py-0" title="Clear query" type="button"><i data-v-21c9158a="" class="bi bi-eraser fs-5"></i></button><button data-v-21c9158a="" class="btn btn-secondary py-0" type="button"><i data-v-21c9158a="" class="bi bi-x fs-5" aria-label="hide query filter"></i></button>
    </div>
    <div data-v-21c9158a="" class="input-group my-1"><select data-v-21c9158a="" class="form-select search-facet-field">
        <option data-v-21c9158a="" selected="" disabled="" value="">Select field...</option>
        <option data-v-21c9158a="" value="acknowledgments__name">Acknowledgment Author</option>
        <option data-v-21c9158a="" value="affects__trackers__ps_update_stream">Affect Update Stream</option>
        <option data-v-21c9158a="" value="affects__affectedness">Affected Affectedness</option>
        <option data-v-21c9158a="" value="affects__ps_component">Affected Component</option>
        <option data-v-21c9158a="" value="affects__ps_module">Affected Module</option>
        <option data-v-21c9158a="" value="cve_description">CVE Description</option>
        <option data-v-21c9158a="" value="requires_cve_description">CVE Description Review</option>
        <option data-v-21c9158a="" value="cve_id">CVE ID</option>
        <option data-v-21c9158a="" value="source">CVE Source</option>
        <option data-v-21c9158a="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-21c9158a="" value="cvss_scores__vector">CVSS Vector</option>
        <option data-v-21c9158a="" value="cwe_id">CWE ID</option>
        <option data-v-21c9158a="" value="embargoed">Embargoed</option>
        <option data-v-21c9158a="" value="affects__trackers__errata__advisory_name">Errata Advisory Name</option>
        <option data-v-21c9158a="" value="workflow_state">Flaw State</option>
        <option data-v-21c9158a="" value="impact">Impact</option>
        <option data-v-21c9158a="" value="major_incident_state">Incident State</option>
        <option data-v-21c9158a="" value="mitigation">Mitigation</option>
        <option data-v-21c9158a="" value="owner">Owner</option>
        <option data-v-21c9158a="" value="statement">Statement</option>
        <option data-v-21c9158a="" value="title">Title</option>
        <option data-v-21c9158a="" value="affects__trackers__external_system_id">Tracker External System ID</option>
        <option data-v-21c9158a="" value="uuid">UUID</option>
      </select><input data-v-21c9158a="" type="text" class="form-control" disabled="">
      <!--v-if-->
      <!--v-if--><button data-v-21c9158a="" class="btn btn-primary py-0" type="button"><i data-v-21c9158a="" class="bi-x fs-5" aria-label="remove field"></i></button>
    </div>
    <div data-v-21c9158a="" class="d-flex mt-2"><button data-v-21c9158a="" class="btn btn-primary me-2" type="submit" aria-label="Advance search button">
        <!--v-if--> Search
      </button><button data-v-21c9158a="" class="btn btn-primary me-2" aria-label="Save filters as default" type="button"> Save as Default </button>
      <!--v-if--><select data-v-21c9158a="" class="sort-by-select form-select search-facet-field" title="When using this sorting with active table column sort, the table column field will be used as secondary sorting key">
        <option data-v-21c9158a="" selected="" disabled="" label="Sort By..."></option>
        <option data-v-21c9158a="" hidden="" label="" value=""></option>
        <option data-v-21c9158a="" value="cve_description">CVE Description</option>
        <option data-v-21c9158a="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-21c9158a="" value="cwe_id">CWE ID</option>
        <option data-v-21c9158a="" value="major_incident_state">Incident State</option>
        <option data-v-21c9158a="" value="mitigation">Mitigation</option>
        <option data-v-21c9158a="" value="source">CVE Source</option>
        <option data-v-21c9158a="" value="statement">Statement</option>
      </select>
      <!--v-if-->
    </div>
  </form>
</details>
<transition-stub data-v-a347fd2c="" name="modal" appear="false" persisted="false" css="true">
  <!--v-if-->
</transition-stub>
<transition-stub data-v-a347fd2c="" name="modal-bg" appear="false" persisted="false" css="true">
  <!--v-if-->
</transition-stub>"
`;

exports[`issueSearchAdvanced > should render when \`isLoading\` is true 1`] = `
"<details data-v-21c9158a="" open="" class="osim-advanced-search-container container-fluid">
  <summary data-v-21c9158a="" class="mb-1">Advanced Search</summary>
  <form data-v-21c9158a="" class="mb-2">
    <div data-v-21c9158a="" class="input-group my-1">
      <div data-v-21c9158a="" type="button" class="query-input form-control bg-secondary text-white"><span data-v-21c9158a="" class="my-auto">Query Filter</span><button data-v-21c9158a="" class="btn btn-sm text-white" type="button"><i data-v-21c9158a="" class="bi bi-question-circle-fill fs-5" aria-label="hide query filter"></i></button></div><textarea data-v-21c9158a="" id="query" type="text" class="form-control"></textarea><button data-v-21c9158a="" disabled="" class="btn btn-sm btn-secondary rounded-0 py-0" title="Clear query" type="button"><i data-v-21c9158a="" class="bi bi-eraser fs-5"></i></button><button data-v-21c9158a="" class="btn btn-secondary py-0" type="button"><i data-v-21c9158a="" class="bi bi-x fs-5" aria-label="hide query filter"></i></button>
    </div>
    <div data-v-21c9158a="" class="input-group my-1"><select data-v-21c9158a="" class="form-select search-facet-field">
        <option data-v-21c9158a="" selected="" disabled="" value="">Select field...</option>
        <option data-v-21c9158a="" value="acknowledgments__name">Acknowledgment Author</option>
        <option data-v-21c9158a="" value="affects__trackers__ps_update_stream">Affect Update Stream</option>
        <option data-v-21c9158a="" value="affects__affectedness">Affected Affectedness</option>
        <option data-v-21c9158a="" value="affects__ps_component">Affected Component</option>
        <option data-v-21c9158a="" value="affects__ps_module">Affected Module</option>
        <option data-v-21c9158a="" value="cve_description">CVE Description</option>
        <option data-v-21c9158a="" value="requires_cve_description">CVE Description Review</option>
        <option data-v-21c9158a="" value="cve_id">CVE ID</option>
        <option data-v-21c9158a="" value="source">CVE Source</option>
        <option data-v-21c9158a="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-21c9158a="" value="cvss_scores__vector">CVSS Vector</option>
        <option data-v-21c9158a="" value="cwe_id">CWE ID</option>
        <option data-v-21c9158a="" value="embargoed">Embargoed</option>
        <option data-v-21c9158a="" value="affects__trackers__errata__advisory_name">Errata Advisory Name</option>
        <option data-v-21c9158a="" value="workflow_state">Flaw State</option>
        <option data-v-21c9158a="" value="impact">Impact</option>
        <option data-v-21c9158a="" value="major_incident_state">Incident State</option>
        <option data-v-21c9158a="" value="mitigation">Mitigation</option>
        <option data-v-21c9158a="" value="owner">Owner</option>
        <option data-v-21c9158a="" value="statement">Statement</option>
        <option data-v-21c9158a="" value="title">Title</option>
        <option data-v-21c9158a="" value="affects__trackers__external_system_id">Tracker External System ID</option>
        <option data-v-21c9158a="" value="uuid">UUID</option>
      </select><input data-v-21c9158a="" type="text" class="form-control" disabled="">
      <!--v-if-->
      <!--v-if--><button data-v-21c9158a="" class="btn btn-primary py-0" type="button"><i data-v-21c9158a="" class="bi-x fs-5" aria-label="remove field"></i></button>
    </div>
    <div data-v-21c9158a="" class="d-flex mt-2"><button data-v-21c9158a="" class="btn btn-primary me-2" type="submit" disabled="" aria-label="Advance search button">
        <div data-v-21c9158a="" class="spinner-border spinner-border-sm"></div> Search
      </button><button data-v-21c9158a="" class="btn btn-primary me-2" aria-label="Save filters as default" type="button" disabled=""> Save as Default </button>
      <!--v-if--><select data-v-21c9158a="" class="sort-by-select form-select search-facet-field" title="When using this sorting with active table column sort, the table column field will be used as secondary sorting key">
        <option data-v-21c9158a="" selected="" disabled="" label="Sort By..."></option>
        <option data-v-21c9158a="" hidden="" label="" value=""></option>
        <option data-v-21c9158a="" value="cve_description">CVE Description</option>
        <option data-v-21c9158a="" value="cvss_scores__score">CVSS Score</option>
        <option data-v-21c9158a="" value="cwe_id">CWE ID</option>
        <option data-v-21c9158a="" value="major_incident_state">Incident State</option>
        <option data-v-21c9158a="" value="mitigation">Mitigation</option>
        <option data-v-21c9158a="" value="source">CVE Source</option>
        <option data-v-21c9158a="" value="statement">Statement</option>
      </select>
      <!--v-if-->
    </div>
  </form>
</details>
<transition-stub data-v-a347fd2c="" name="modal" appear="false" persisted="false" css="true">
  <!--v-if-->
</transition-stub>
<transition-stub data-v-a347fd2c="" name="modal-bg" appear="false" persisted="false" css="true">
  <!--v-if-->
</transition-stub>"
`;
